### ISSUE TITLE
[Regression] EditorConfig templates are missing their display name

### ISSUE THREAD

**Issue by davkean** [dotnet/templates#167]
**Version Used**: 
Version 16.3.0 Preview 3.0 [29221.207.master]

**Steps to Reproduce**:

**Expected Behavior**:
![image](https://user-images.githubusercontent.com/1103906/63518698-f0c62700-c534-11e9-8faa-5b39c265306d.png)

**Actual Behavior**:

16.2:
![image](https://user-images.githubusercontent.com/1103906/63518757-10f5e600-c535-11e9-88bf-89cf5a627927.png)

**Comment by jasonmalinowski** [comment]
@davkean @tmeschter So who does this bug get assigned to then?

**Comment by davkean** [comment]
Who owns this?

**Comment by jasonmalinowski** [comment]
I was hoping you knew. This is a repo I didn't really know existed. :-)

**Comment by tmeschter** [comment]
Project system owns this.

**Comment by tmeschter** [comment]
@jmarolf Any idea what might be wrong here?

**Comment by jmarolf** [comment]
dunno, I haven't changed anything in [setup](https://github.com/dotnet/templates/tree/master/src/Setup/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig.Setup) for over a year.  For item templates to show up in the default items category and work, they need to be copied to the `Common7\IDE\NewFileItems` folder. The only way I know how to do that is with a vsdir file.

So the name for the item template should either be coming from [this vsdir](https://github.com/dotnet/templates/blob/master/src/Setup/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig.Setup/NewEditorconfigFileItem.vsdir) or this [template file](https://github.com/dotnet/templates/blob/master/src/Templates/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig/Editorconfig.CS.Default.vstemplate#L3).

Can we confirm that this package from [here ](https://github.com/dotnet/templates/blob/4aaa0451f27ed8f8e7bd42fbb01adce18336878d/src/Setup/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig.Wizard.Setup/ResourcePackage.pkgdef)is being installed/deployed?

I expect this resource string to load:
https://github.com/dotnet/templates/blob/4aaa0451f27ed8f8e7bd42fbb01adce18336878d/src/Setup/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig.Wizard.Setup/VSPackage.resx#L127

**Comment by davkean** [comment]
I compared the working version against the new version and they are identical apart from a version number changing 15.8.0.0 -> 16.3.0.0 in the dll. The package (looks) to be installed.

![image](https://user-images.githubusercontent.com/1103906/63629961-73480700-c659-11e9-94fd-ab7ae79dd37a.png)

**Comment by davkean** [comment]
Can other folks repro this?

**Comment by jmarolf** [comment]
@davkean 	I can, I'll take a look on monday.  There must be something I am not seeing that has changed

**Comment by jmarolf** [comment]
This bug is caused by this setting not being respected during compilation:

What is in the file:
https://github.com/dotnet/templates/blob/e785ec0793b93173d8abdca462d889487a458e9f/src/Setup/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig.Wizard.Setup/ResourcePackage.pkgdef#L10

What gets installed:
```
"Assembly"="Microsoft.VisualStudio.Templates.Editorconfig.Wizard.Setup, Version=16.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"
```

**Comment by jmarolf** [comment]
I assume that this is because this target does not exist in arcade?

https://github.com/dotnet/templates/blob/e785ec0793b93173d8abdca462d889487a458e9f/src/Setup/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig.Wizard.Setup/Microsoft.VisualStudio.Templates.Editorconfig.Wizard.Setup.csproj#L57-L61

@tmat 	what is the correct way to include a `pkgdef` file in arcade?

**Comment by jasonmalinowski** [comment]
The PkgdefOutputGroup is only needed if you're doing generation automatically; if you're manually specifying it then you can just directly state it your .vsixmanifest.

If you're seenig it switch from Codebase to Assembly, that leads me to believe that either:

1. You're not specifying UseCodebase somewhere and want generation, or
2. You don't want generation at all, but are for some reason unintentionally using generation.

**Comment by dmonroym** [comment]
@jasonmalinowski Is there any reason for this behavior to have changed if this code's not been changed in a while?

With regard to your comments:
> You're not specifying UseCodebase somewhere and want generation

From what I'm seeing if I try specifying UseCodebase I get the following error: `UseCodebase should not be true for VSIX project that turns NGEN on`

>You don't want generation at all, but are for some reason unintentionally using generation.

Is there any way to validate if we're using generation?

**Comment by jmarolf** [comment]
>  Is there any reason for this behavior to have changed if this code's not been changed in a while?

This is likely an arcade change. 

We generate a `pkgdef` for a VSIX and include an manually created pkgdef because solution items are not supported by the VS SDK.

I assume this is not supported in arcade

**Comment by jasonmalinowski** [comment]
So things being loaded by a CodeBase causes issues with ngen images; Roslyn for example doesn't have a Codebase entry in our Packages key, it does point to the assembly name. Then there's *also* an attribute that says when loading our assembly where to find it. That explains Arcade's message: if you're ngenning, you need to use the Assembly one and have the additional entry.

> manually created pkgdef because solution items are not supported by the VS SDK.

What do you mean? Your pkgdef is just defining a generic package, there's nothing special about it. I'm not sure why you wouldn't be generating it...

**Comment by jmarolf** [comment]
Looking at this, its likely this was always wrong and it was only happenstance that the correct pkgdef was selected. @jasonmalinowski what is the right attribute / assets specification to get this to be generated?

https://github.com/dotnet/templates/blob/d62058cc0dd0f18e1dbe2ea2da4f4d7042f18af5/src/Setup/Editorconfig/Microsoft.VisualStudio.Templates.Editorconfig.Wizard.Setup/ResourcePackage.pkgdef#L6-L11

**Comment by davidwengier** [comment]
Fixed in #167
